### PR TITLE
QA 466: Launch wf w/ wrong root entity and on set with incorrect expression

### PR DESCRIPTION
**Pull request by ansingh7115**
- [ ] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Check that the **Product Owner** has signed off on any user-facing changes
- [ ] **Submitter**: Make sure Swagger is updated if API changes
  - [ ] **...and Orchestration's Swagger too!**
- [ ] **Submitter**: If updating admin endpoints, also update [firecloud-admin-cli](https://github.com/broadinstitute/firecloud-admin-cli)
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [ ] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [ ] **Submitter**: Database checks:
  * If PR includes new or changed db queries, include the explain plans in the description
  * Make sure liquibase is updated if appropriate
  * If doing a migration, take a backup of the
  [dev](https://console.cloud.google.com/sql/instances/terraform-qfarbdq3lrexxck5htofjs5z6m/backups?project=broad-dsde-dev&organizationId=548622027621)
  and
  [alpha](https://console.cloud.google.com/sql/instances/terraform-r4caezzc35c4tb7pgdhwkmme4y/backups?project=broad-dsde-alpha&organizationId=548622027621)
  DBs in Google Cloud Console
- [ ] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [ ] Tell your tech lead (TL) that the PR exists if they want to look at it
- [ ] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Assign to submitter** to finalize
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via Slack and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

**Comment by ansingh7115** [comment]
jenkins retest

File: automation/src/test/scala/org/broadinstitute/dsde/test/api/MethodLaunchSpec.scala [hunk 1]
```diff
@@ -33,8 +37,11 @@ class MethodLaunchSpec extends TestKit(ActorSystem("MySpec")) with FreeSpecLike
...
           val method = MethodData.InputRequiredMethod.copy(methodName = methodName)
+          val methodConfigName = createMethodConfigName
+
           Rawls.methodConfigs.createMethodConfigInWorkspace(billingProject, workspaceName, method,
-            method.methodNamespace, methodConfigName, 1, Map.empty, Map.empty, method.rootEntityType)
+            method.methodNamespace, methodConfigName, 1, Map.empty, SimpleMethodConfig.outputs, method.rootEntityType)
+
           val exception = intercept[RestException](Rawls.submissions.launchWorkflow(billingProject, workspaceName, method.methodNamespace, methodConfigName, "participant",
           "participant1", "this", false))
           exception.message.parseJson.asJsObject.fields("message").convertTo[String].contains("Missing inputs:") shouldBe true
```
Comment (aweng98) [inline automation/src/test/scala/org/broadinstitute/dsde/test/api/MethodLaunchSpec.scala:47]: do you think using `eventually` when intercept exception message makes sense to have in tests? I hate tests running in alpha env bc it is notoriously slow.

**Review by aweng98 — APPROVED** [review]
LGTM

File: automation/src/test/scala/org/broadinstitute/dsde/test/api/MethodLaunchSpec.scala (hunk 1 above)
Comment (ansingh7115) [reply automation/src/test/scala/org/broadinstitute/dsde/test/api/MethodLaunchSpec.scala:47]: Not in the cases where we expect an exception that originates from rawls. We would have an `eventually` if it was something we'd have to wait for, such as a response from Cromwell, where we might need to wait in some kind of queue and then have to poll. But for a launch workflow call that error at rawls before submitting to cromwell, we expect the exception to be thrown immediately on that call, not eventually after some amount of polling. If it is not thrown immediately, then something is wrong and we want the test to fail.

**Review by marctalbott — APPROVED** [review]

**Pull request merged**